src/interactive-auth.js

```diff
@@ -168,15 +257,64 @@ InteractiveAuth.prototype = {
         if (!nextStage) {
             throw new Error("No incomplete flows from the server");
         }
+        if (nextStage == this._currentStage) {
+            // we've already started: don't re-start it
+            return;
+        }
+        this._currentStage = nextStage;
 
-        let stageError = null;
         if (this._data.errcode || this._data.error) {
-            stageError = {
+            this._stateUpdatedCallback(nextStage, {
                 errcode: this._data.errcode || "",
                 error: this._data.error || "",
-            };
+            });
+            return;
+        }
+
+        const stageStatus = {};
+        if (nextStage == EMAIL_STAGE_TYPE) {
+            stageStatus.busy = true;
+        }
+        this._stateUpdatedCallback(nextStage, stageStatus);
+
+        // Do stage-specific things to start the stage. These would be
+        // an obvious thing to stick in a different file / function if there
+        // were more of them.
+        if (nextStage == EMAIL_STAGE_TYPE) {
+            if (this._emailSid) {
+                this.poll();
+            } else {
+                this._requestEmailToken().catch(
+                    this._completionDeferred.reject,
+                ).finally(() => {
+                    this._stateUpdatedCallback(nextStage, { busy: false });
```

Comment:
oh wait, what's happening here? why aren't we calling submitAuthDict ourselves here?